woocommerce/woocommerce-ios
# Blaze: Unify intro modal logic in Blaze campaign creation coordinator

Closes: #11810 


## Description

This PR adds the logic to navigate to the Blaze intro modal to the Blaze campaign creation coordinator, and reuse this for the three entry points My Store, Product Form, Campaign List.

Also - removed the outdated Blaze source `productMoreMenu`.

## Testing instructions

With `blazei3NativeCampaignCreation` feature flag on.
- Log in to a store eligible for Blaze but without any campaign created in the past.
- Tap the product row in the Blaze section on My Store screen, notice that the intro screen is displayed. `blaze_entry_point_displayed` should be logged in Xcode console with source `intro_view`.
- Tap the Create your campaign button, event `blaze_entry_point_tapped` should be logged with source `intro_view`. The creation flow should be started with the product.
- Switch to the Products tab and select any published item.
- Tap Promote with Blaze, the intro modal should be displayed with correct track events.
- Switch to the Menu tab and select Blaze. The intro modal should be displayed only once upon opening the screen.
- Tap the Create button on the campaign list. The intro modal should not be displayed.

Please feel free to test again with a store with existing campaigns or with the feature flag off. Everything should work as before.

## Screenshots


https://github.com/woocommerce/woocommerce-ios/assets/5533851/b28b9053-dd48-4f87-beb0-191ba6bf9287



---
